skbuild/src/harness.sk

```diff
@@ -322,18 +317,37 @@ fun ensureDirectory(path: String, verbose: Bool): void {
   }
 }
 
-fun arch(): String {
-  p = System.subprocess(Array["gcc", "-dumpmachine"]);
-  p.stdout.trim();
-}
-
-fun copyObjOptions(): (String, String) {
-  arch().split("-").maybeGet(0) match {
-  | Some("x86_64") -> ("i386:x86-64", "elf64-x86-64")
-  | Some("aarch64")
-  | _ ->
-    ("aarch64", "elf64-littleaarch64")
-  }
+fun copyBinary(name: String, src: String, target: String): void {
+  srcFile = IO.File::open(src, IO.OpenOptions{read => true});
+  dstFile = IO.File::open(
+    target,
+    IO.OpenOptions{write => true, create => true},
+  );
+  _ = dstFile.write(`unsigned char ${name}[] = {`);
+  count = 0;
+  loop {
+    code = try {
+      read_byte(srcFile.fileno)
+    } catch {
+    | EndOfFile _ -> -1
+    | exn -> throw exn
+    };
+    if (code == -1) {
+      break void
+    };
+    if (count == 0) {
+      _ = dstFile.write("\n  ");
+    } else if (count % 12 == 0) {
+      _ = dstFile.write(",\n  ");
+    } else {
+      _ = dstFile.write(",  ");
+    };
+    _ = dstFile.write("0x" + Chars.intToHexDigits(code, 2));
+    !count = count + 1;
+  };
+  _ = dstFile.write(`\n};\nunsigned int ${name}_len = ${count};\n`);
+  srcFile.close();
+  dstFile.close()
 }
```

Comment:
@beauby, this is the function I mentioned would be a good candidate to see if your efficient IO APIs are comfortable to use.